packages/ses/src/compartment-shim.js

```diff
@@ -72,7 +101,45 @@ export class Compartment {
 
     assign(globalObject, endowments);
 
+    // Map<FullSpecifier, ModuleCompartmentRecord>
+    const moduleRecords = new Map();
+    // Map<FullSpecifier, ModuleInstance>
+    const instances = new Map();
+    // Map<FullSpecifier, Alias{Compartment, FullSpecifier}>
+    const aliases = new Map();
+    // Map<FullSpecifier, {ExportsProxy, ProxiedExports, activate()}>
+    const deferredExports = new Map();
+
+    for (const [specifier, module] of entries(modules)) {
+      if (typeof module === 'string') {
+        throw new TypeError(
+          `Cannot map module ${q(specifier)} to ${q(
+            module,
+          )} in parent compartment`,
+        );
+      } else {
+        const alias = moduleAliases.get(module);
+        if (alias != null) {
+          // Modules from other components.
+          aliases.set(specifier, alias);
+        } else {
+          // TODO create and link a synthetic module instance from the given namespace object.
+          throw ReferenceError(
+            `Cannot map module ${q(
+              specifier,
+            )} because it has no known compartment in this realm`,
+          );
+        }
+      }
+    }
+
     privateFields.set(this, {
+      resolveHook,
+      importHook,
+      aliases,
+      moduleRecords,
+      deferredExports,
+      instances,
       globalTransforms,
       globalObject,
     });
```

Comment:
Might as well rename the getter on line 148 below from `global` to `globalThis` in this PR. I don't see any good reason to separate that into another PR.

Reply:
This has turned out to be more invasive than I expected and I would like to capture this in a subsequent change.